File: src/color.ts
export class Color {

	private red: number;
	private green: number;
	private blue: number;
	private index?: number;

	constructor(red: number, green: number, blue: number) {
		this.red = red;
		this.green = green;
		this.blue = blue;
	}

	public toString(): string {
		return JSON.stringify(this.toJson());
	};

	public toJson(): Object {
		return {
			red: this.red,
			green: this.green,
			blue: this.blue
		}
	}

}

File: Test/Color.cs
namespace Test
{
	class Color
	{
		private int red;
		private int green;
		private int blue;

		public Color(int red, int green, int blue)
		{
			this.red = red;
			this.green = green;
			this.blue = blue;
		}

		public string toJsonString()
		{
			return $"{{r: {this.red}, g: {this.green}, b: {this.blue}}}";
		}
	}
}


